source/contracts/trading/ShareToken.sol

```diff
@@ -36,6 +36,10 @@ contract ShareToken is DelegationTarget, Extractable, ITyped, Initializable, Var
         return true;
     }
 
+    function trustedTransfer(address _source, address _destination, uint256 _attotokens) public onlyWhitelistedCallers onlyInGoodTimes afterInitialized returns (bool) {
```

Comment:
Bleh.  I'm generally not a fan of the whole "whitelisted caller" thing because it makes auditing _really_ hard as you have to assume that _any_ contract in the whitelist can call it.  Also, it means that the Augur dev team has the ability to transfer user assets without constraint.  It is possible that whitelisted callers already has the ability to steal all user funds, in which case this doesn't matter much but if not then this is a pretty major break in the security model (the one currently going through audits).

I would be more OK with this if we constrained it to specific contracts that are allowed to transfer funds (using something like `controller.assertOnlySpecifiedCaller`).  In general, anytime I see `onlyWhitelistedCallers` I become very apprehensive for the above reasons.